File: findErrorNums/mySecondSolution.java
// link: https://leetcode.com/problems/set-mismatch/

class Solution {
    public int[] findErrorNums(int[] nums) {
        Arrays.sort(nums);
        
        int extraNum = Integer.MAX_VALUE,
            missedNum = Integer.MIN_VALUE;
        
        for (int index = 0; index < nums.length; ++index) {
            if (index == 0) {
                if (nums[index] != 1)
                    missedNum = 1;
            }
            else {
                if (nums[index] == nums[index - 1])
                    extraNum = nums[index];
                else if (nums[index] - nums[index - 1] > 1)
                    missedNum = nums[index] - 1;
            }
        }
        
        if (nums[nums.length - 1] < nums.length)
            missedNum = nums.length;
        
        int ans[] = {extraNum, missedNum};
        
        return ans;
    }
}

File: count_bits/my_second_solution.py
# https://leetcode.com/problems/counting-bits/
from typing import List


class Solution:
    def countBits(self, n: int) -> List[int]:
        if n == 0:
            return [0]

        ans = [0 for _ in range(n+1)]
        ans[1] = 1

        for num in range(2, n+1):
            if num % 2 == 0:
                ans[num] = ans[num // 2]
            else:
                ans[num] = ans[num - 1] + 1

        return ans
